src/Dictionaries/getDictionaryConfigurationFromAST.cpp

```diff
@@ -484,8 +484,13 @@ void buildConfigurationFromFunctionWithKeyValueArguments(
             /// It's not possible to have a function in a dictionary definition since 22.10,
             /// because query must be normalized on dictionary creation. It's possible only when we load old metadata.
             /// For debug builds allow it only during server startup to avoid crash in BC check in Stress Tests.
```

Comment:
The comment above needs to be adjusted. I'm not sure if we even need the conditionals anymore. I think that at this point it's ok to break compatibility with 22.10 so we could check if unconditionally. WDTY?